hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -917,7 +917,11 @@ private Object putFromLoadInternal(Data key, Object value, long ttl, long maxIdl
             value = mapServiceContext.interceptPut(name, null, value);
             record = createRecord(value, ttl, maxIdle, now);
             storage.put(key, record);
-            mutationObserver.onPutRecord(key, record);
+            if (canPublishLoadEvent()) {
```

Comment:
Here we need this condition, see line 940.

Reply:
Probably this check could be removed since the event journal is private to jet and they don't do rolling upgrades -> we know all members are 3.11 and we don't need to check the cluster version. But let's keep it in for now. As a quick addition, can you add the `// RU_COMPAT_3_10` to the `com.hazelcast.map.impl.recordstore.DefaultRecordStore#canPublishLoadEvent` method?